src/lib/coda_lib/conf_dir.ml

```diff
@@ -32,11 +32,52 @@ let check_and_set_lockfile ~logger conf_dir =
             "Could not create the daemon lockfile" ("lockfile", lockfile)
             [%sexp_of: string * string]
           |> Error.raise )
-  | `Yes ->
-      Mina_user_error.raisef
-        "A daemon is already running with the current configuration directory \
-         (%s), or you may need to remove the lockfile (%s)"
-        conf_dir lockfile
+  | `Yes -> (
+      let open Async in
+      match%map
+        Monitor.try_with ~extract_exn:true (fun () ->
+            Reader.with_file ~exclusive:true lockfile ~f:(fun reader ->
+                let%bind pid =
+                  let rm_and_raise () =
+                    Core.Unix.unlink lockfile ;
+                    Mina_user_error.raise
+                      "Invalid format in lockfile (removing it)"
+                  in
+                  match%map Reader.read_line reader with
+                  | `Ok s -> (
+                    try Pid.of_string s with _ -> rm_and_raise () )
+                  | `Eof ->
+                      rm_and_raise ()
+                in
+                let still_running =
+                  match Signal.(send zero) (`Pid pid) with
```

Comment:
Nit: would be nice to have a comment here `Sig=0 does not send any signal, only performs error checking (See man kill(1))`